internal/database/network.go

```diff
@@ -457,37 +595,49 @@ func SearchNetworks(conditions map[string]string, properties map[string]interfac
 	return
 }
 
-func ReplaceAllNetworks(networks []sls_common.Network) (err error) {
-	trans, beginErr := DB.Begin()
+func ReplaceAllNetworksContext(ctx context.Context, networks []sls_common.Network) error {
+	tx, beginErr := DB.BeginTx(ctx, nil)
 	if beginErr != nil {
-		err = errors.Errorf("unable to begin transaction: %s", beginErr)
-		return
+		return errors.Errorf("unable to begin transaction: %s", beginErr)
 	}
 
-	version, err := IncrementVersion(trans, "replaced all networks")
+	err := ReplaceAllNetworks(tx, networks)
+	if err != nil {
+		tx.Rollback()
+		return err
+	}
+
+	commitErr := tx.Commit()
+	if commitErr != nil {
+		return errors.Errorf("unable to commit transaction: %s", commitErr)
+	}
+
+	return nil
+}
+
+func ReplaceAllNetworks(tx *sql.Tx, networks []sls_common.Network) (err error) {
+	version, err := IncrementVersion(tx, "replaced all networks")
 	if err != nil {
 		err = errors.Errorf("insert to version_history failed: %s", err)
-		_ = trans.Rollback()
+		_ = tx.Rollback()
```

Comment:
Rollback will get called twice. Maybe remove this one.
I'm not sure if this is a problem.
The other other rollbacks in this function were removed.

Reply:
Done